kernels/optimized/cpu/binary_ops.h

```diff
@@ -48,8 +49,38 @@ enum class ElementwiseOptimizedPath {
   kBroadcastLastDimReverseArguments,
 };
 
+enum class BinaryOpType {
+  kAdd,
+  kSub,
+  kMul,
+  kDiv,
+};
+
 namespace internal {
 
+template <BinaryOpType op_type>
+struct BinaryOpTypeName;
+
+template <>
+struct BinaryOpTypeName<BinaryOpType::kAdd> {
+  static constexpr char kName[] = "add.out";
+};
+
+template <>
+struct BinaryOpTypeName<BinaryOpType::kSub> {
+  static constexpr char kName[] = "sub.out";
+};
+
+template <>
+struct BinaryOpTypeName<BinaryOpType::kMul> {
+  static constexpr char kName[] = "mul.out";
+};
+
+template <>
+struct BinaryOpTypeName<BinaryOpType::kDiv> {
+  static constexpr char kName[] = "div.out";
+};
+
```

Comment:
you don't need to do this. see existing example: https://github.com/pytorch/executorch/blob/c82a7df29e25b1fb348d8e833d82fe00d95ce45d/kernels/portable/cpu/op_rsub.cpp#L50-L55

the secret sauce is that the string literal has to be a static constexpr const char [] and then you can pass it to a `const char*` template argument directly.

Reply:
Thanks. I was hoping you would point me to something better for this